aspects/schema_test.go

```diff
@@ -1892,3 +1829,60 @@ func (*schemaSuite) TestValidationError(c *C) {
 		c.Assert(err.Error(), Equals, fmt.Sprintf(`cannot accept element in %q: base error`, tc.expected))
 	}
 }
+
+func (*schemaSuite) TestUnexpectedTypes(c *C) {
+	type testcase struct {
+		schemaType   string
+		expectedType string
+	}
+
+	tcs := []testcase{
+		{
+			schemaType:   `{"type": "array", "values": "any"}`,
+			expectedType: "array",
+		},
+		{
+			schemaType:   `{"type": "map", "values": "any"}`,
+			expectedType: "map",
+		},
+		{
+			schemaType:   `"int"`,
+			expectedType: "int",
+		},
+		{
+			schemaType:   `"number"`,
+			expectedType: "number",
+		},
+		{
+			schemaType:   `"string"`,
+			expectedType: "string",
+		},
+		{
+			schemaType:   `"bool"`,
+			expectedType: "bool",
+		},
+	}
+
+	for _, tc := range tcs {
+		schemaStr := []byte(fmt.Sprintf(`{
+	"schema": {
+		"foo": %s
+	}
+}`, tc.schemaType))
+		schema, err := aspects.ParseSchema(schemaStr)
+		c.Assert(err, IsNil)
+
+		var testValue, testType string
+		if tc.expectedType != "bool" {
```

Comment:
nitpick: I think the test would be a little easier to read at the cost of some duplication if this `if` were removed and `testValue` and `testType` were put in each `testcase`

Reply:
I removed the if/else and added an `interface{}` field so we can get both the value and type from it. Thanks